modules/apps/frontend-js/frontend-js-clay-sample-web/src/main/resources/META-INF/resources/js/App.js

```diff
@@ -12,20 +12,73 @@
  * details.
  */
 
-import ClayAlert from '@clayui/alert';
+import ClayCard from '@clayui/card';
+import ClayForm, {ClayInput} from '@clayui/form';
+import {useLiferayState} from '@liferay/frontend-js-react-web';
+import {State} from '@liferay/frontend-js-state-web';
 import React from 'react';
 
 import '../css/main.scss';
 
-export default () => {
+// Shared state (atoms and selectors); normally these would be in a separate
+// file.
+
+const userAtom = State.atom('clay-sample-atom', {
+	name: Liferay.ThemeDisplay.getUserName(),
+});
+
+const userSelector = State.selector('clay-sample-selector', (get) => {
+	const user = get(userAtom);
+
+	return `${user.name} (${user.name.length})`;
+});
+
+// Components that access that shared state:
+
+function Name() {
+	const [userNameAndLength] = useLiferayState(userSelector);
+
 	return (
-		<div>
-			<ClayAlert title="Info">
-				This widget is used to test out Clay components. Simply add
-				whatever JS you want to App.js and redeploy.
```

Comment:
Now that I see this... the whole point of this module (which shouldn't be released) was for QA to literally "add whatever JS you want to App.js and redeploy". At least that's how @bryceosterhaus and @john-co envisioned it if I remember correctly.

I like the idea of having a state sample, just not sure what could be a good place for it. The best that comes to mind is the `frontend-taglib-clay-sample-web` or if we want, maybe `StateApp.js` in here? 🤷

Reply:
In my rebase + squash (not pushed yet) I dropped this commit and parked it [over here in a gist](https://gist.github.com/wincent/2d1e822e6a3478488a5036fcaff1f30b). It's possible we won't need a sample once we have a few real usages to point at in liferay-portal.

I've been using this module in a few PRs in the way that you say:

> for QA to literally "add whatever JS you want to App.js and redeploy"

but I always remove it before the final version.

Reply:
Got it, thanks! 👍